### ISSUE TITLE
Correctly detect noqa when used in method calls

### ISSUE THREAD

**Issue by yabirgb** [charliermarsh/ruff#599]
### Problem

At rotki we have places where the `noqa: E501` rule comes in after or in function declarations like https://github.com/rotki/rotki/blob/535b7995dbf65f2824fac6fb8b73c1e1f2350613/rotkehlchen/db/upgrades/v33_v34.py#L113

### Example

In those places flake correctly detects the rule to skip the error but ruff doesn't. For a reproducible scenario I have 

```python
def caller(string: str) -> None:
    assert string is not None


caller(
    """
    This is test for a very long string where flake would prevent raising the error sicne we use noqa but ruff doesn't
    """,  # noqa: E501
)
```

That has this two different outputs

```
(rotki-main) ➜  rotki git:(trueblocks) ✗ flake8 ruff_test.py
(rotki-main) ➜  rotki git:(trueblocks) ✗ ruff ruff_test.py 
Found 1 error(s).
ruff_test.py:7:1: E501 Line too long (118 > 100 characters)
(rotki-main) ➜  rotki git:(trueblocks) ✗ 
```

### Expected behaviour

ruff should detect that the noqa rule is being used to skip the long line

**Comment by charliermarsh** [comment]
Thanks for this! Great issue. Will take a look soon.

**Comment by charliermarsh** [comment]
(Reproduced.)

**Comment by charliermarsh** [comment]
Fixed! This will go out later today.
